File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int canMove = 0;
        for (int num : nums) {
            if (canMove < 0) return false;
            canMove = Math.max(canMove, num) - 1;
        }
        return true;
    }
}

File: 0055-jump-game/0055-jump-game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    const jumpMap = Array(nums.length).fill(false);
    jumpMap[0] = true;
    
    for (let i = 0; i < nums.length; i++) {
        if (!jumpMap[i]) continue;
        for (let j = 1; j <= nums[i]; j++) {
            if (i + j >= nums.length) break;
            jumpMap[i + j] = true;
        }
    }

    return jumpMap[nums.length - 1];
};